projects/models.nix

```diff
@@ -0,0 +1,126 @@
+{
+  lib,
+  pkgs,
+  sources,
+}:
+let
+  yants = import sources.yants { };
+
+  inherit (yants)
+    string
+    list
+    option
+    attrs
+    enum
+    either
+    struct
+    drv
+    path
+    restrict
+    ;
+
+  programType = struct "program" {
+    name = string;
+    source = drv;
+    documentation = optionalStruct {
+      build = option string;
+      tests = option string;
+    };
+    examples = nonEmtpyAttrs (option exampleType);
+  };
+
+  serviceType = struct "service" {
+    name = string;
+    documentation = optionalStruct {
+      config = option string;
+    };
+    examples = nonEmtpyAttrs (option exampleType);
+  };
+
+  exampleType = struct "example" {
+    description = option string; # TODO: should this be non-optional?
+    path = either string path;
```

Comment:
I forgot, what did we say `path` will be used for?

Reply:
The path to the expression we use as an example (see some projects that have examples, e.g. Atomic Data)